tests/kubeless-test-client/test-kubeless.go

```diff
@@ -169,21 +227,39 @@ func publishEvent(testID string) {
 	}
 }
 
-func ensureCorrectLog(namespace, funName string, pattern *regexp.Regexp, match string) {
+func ensureCorrectLog(namespace, funName string, pattern *regexp.Regexp, match string, serviceBinding bool) {
 	timeout := time.After(2 * time.Minute)
 	tick := time.Tick(1 * time.Second)
+	sbuID := ""
+	if serviceBinding {
+		sbuID = getSBUID(namespace, funName)
+	}
 	for {
+		var cmd *exec.Cmd
+		if sbuID == "" {
+			cmd = exec.Command("kubectl", "-n", namespace, "get", "pods", "-l", "function="+funName, "--field-selector=status.phase==Running", "--output=jsonpath={.items[0].metadata.name}")
+		} else {
+			cmd = exec.Command("kubectl", "-n", namespace, "get", "pods", "-l", "use-"+sbuID+","+"function="+funName, "--field-selector=status.phase==Running", "--output=jsonpath={.items[0].metadata.name}")
+		}
+
+		stdoutStderr, err := cmd.CombinedOutput()
+		if err != nil {
+			log.Fatalf("Unable to get running pod for function: %v due to following reason: %v", funName, err)
+		}
+		pod := string(stdoutStderr)
+		log.Printf("Checking logs for pods: %v", pod)
 		select {
 		case <-timeout:
-			cmd := exec.Command("kubectl", "-n", namespace, "log", "-l", "function="+funName, "-c", funName)
+			cmd := exec.Command("kubectl", "-n", namespace, "logs", pod)
```

Comment:
This will now work as there are 2 containers and we would need the name of the container.